File: LovePlay-UIKit-Masonry/LovePlay/Class/Main/Controller/MainTabBarController.m
//
//  MainTabBarController.m
//  LovePlay
//
//  Created by weiying on 2017/1/22.
//  Copyright © 2017年 Yuns. All rights reserved.
//

#import "MainTabBarController.h"
#import "MainNavigationController.h"
#import "NewsListViewController.h"
#import "RecommendViewController.h"
#import "ZoneListViewController.h"

@interface MainTabBarController ()

@end

@implementation MainTabBarController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self configureTabBar];
    [self addNewsViewController];
    [self addRecommendViewController];
    [self addZoneViewController];
}

- (void)configureTabBar
{
    [[UITabBarItem appearanceWhenContainedIn:[self class], nil] setTitleTextAttributes:@{NSForegroundColorAttributeName :[UIColor colorWithRed:113/255.0 green:113/255.0 blue:113/255.0 alpha:1.0] } forState:UIControlStateNormal];
    
    [[UITabBarItem appearanceWhenContainedIn:[self class], nil] setTitleTextAttributes:@{NSForegroundColorAttributeName :[UIColor colorWithRed:218/255.0 green:85/255.0 blue:107/255.0 alpha:1.0] } forState:UIControlStateSelected];
}

- (void)addNewsViewController
{
    NewsListViewController *newsViewController = [[NewsListViewController alloc] init];
    [self addChildViewController:newsViewController title:@"资讯" imageName:@"icon_zx_nomal_pgall" selectImageName:@"icon_zx_pressed_pgall"];
}

- (void)addRecommendViewController
{
    RecommendViewController *recommendViewController = [[RecommendViewController alloc] init];
    [self addChildViewController:recommendViewController title:@"精选" imageName:@"icon_jx_nomal_pgall" selectImageName:@"icon_jx_pressed_pgall"];
}

- (void)addZoneViewController
{
    ZoneListViewController *zoneViewController = [[ZoneListViewController alloc] init];
    [self addChildViewController:zoneViewController title:@"社区" imageName:@"icon_sq_nomal_pgall" selectImageName:@"icon_sq_pressed_pgall"];
}

- (void)addChildViewController:(UIViewController *)childController title:(NSString *)title imageName:(NSString *)imageName selectImageName:(NSString *)selectImageName
{
    childController.title = title;
    childController.tabBarItem.title = title;
    childController.tabBarItem.image = [UIImage imageNamed:imageName];
    childController.tabBarItem.selectedImage = [[UIImage imageNamed:selectImageName] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
    
    MainNavigationController *nav = [[MainNavigationController alloc] initWithRootViewController:childController];
    [self addChildViewController:nav];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: LovePlay-Swift3/LovePlay/Class/Main/Controller/MainTabBarController.swift
//
//  MainTabBarController.swift
//  LovePlay
//
//  Created by weiying on 2017/3/10.
//  Copyright © 2017年 yuns. All rights reserved.
//

import UIKit

class MainTabBarController: UITabBarController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.configureTabBar()
        self.addNewsViewController()
        self.addRecommendViewController()
        self.addZoneViewController()
    }

    func configureTabBar() {

        UITabBarItem.appearance().setTitleTextAttributes([NSForegroundColorAttributeName : RGB(r: 113, g: 113, b: 113)], for: .normal)
        
        UITabBarItem.appearance().setTitleTextAttributes([NSForegroundColorAttributeName : RGB(r: 218, g: 85, b: 107)], for: .selected)
    }
    
    func addNewsViewController(){
        let newsVC = NewsListViewController()
        
        self.addChildIntoTabBarController(childController: newsVC, title: "资讯", imageName: "icon_zx_nomal_pgall", selectImageName: "icon_zx_pressed_pgall")
    }
    
    func addRecommendViewController(){
        let newsVC = RecommendViewController()
        
        self.addChildIntoTabBarController(childController: newsVC, title: "精选", imageName: "icon_jx_nomal_pgall", selectImageName: "icon_jx_pressed_pgall")
    }
    
    func addZoneViewController(){
        let newsVC = ZoneListViewController()
        
        self.addChildIntoTabBarController(childController: newsVC, title: "社区", imageName: "icon_sq_nomal_pgall", selectImageName: "icon_sq_pressed_pgall")
    }
    
    
    func addChildIntoTabBarController(childController : UIViewController, title : String, imageName : String, selectImageName : String) {
        childController.title = title
        childController.tabBarItem.title = title;
        childController.tabBarItem.image = UIImage(named: imageName)
        childController.tabBarItem.selectedImage = UIImage(named: selectImageName)?.withRenderingMode(.alwaysOriginal)
        
        let childNav = MainNavigationController(rootViewController: childController)
        self.addChildViewController(childNav)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

}
